usagov/vote-gov-drupal
# Vote-1116: High Contrast Function - Proof of Concept

## Jira ticket

[Vote-1116](https://cm-jira.usa.gov/browse/VOTE-1116)

## Description

Adding a proof of concept function for high contrast and other future theme updates such as dark mode down the road.

Within this PR there were several updates: 
- Adding the function to toggle though the various themes that are to be added 
- Updating the page to add the data attribute `data-theme` which will be used to add the correct color style 
- Updated one section to show how to add the other color styles and also added said example styles within the variables file

There can and will need to be other work to get the pages and components to be able to cycle through the new styles that are in coming for high contrast.

## Deployment and testing

### Post-deploy steps

1. run `lando retune` and then cd into the `votegov` theme and run `npm run build`


### QA/Testing instructions

1. visit site and verify that the test button is not showing on the page, then log in as an admin and check now that the button is showing.. the expected behavior is that while logged out this should not be seen.
2. click on the `Toggle Theme` button and the background color for the nav bar should go between white and black and should keep the same style going from page to page.
3. open the DOM and verify that the `data-theme` is going between default and contrast.  
4. once you are on contrast visit other pages and verify in the DOM that it is still set.
5. visit the js file and add another item to the array options and verify that the new one is being added as you click on the button.

## Checklist for the Developer




## Checklist for the Peer Reviewers
